core/multigrid/amgx_pgm.cpp

```diff
@@ -153,9 +155,9 @@ void AmgxPgm<ValueType, IndexType>::generate()
     // TODO: use less memory footprint to improve it
     auto coarse_matrix =
         share(matrix_type::create(exec, gko::dim<2>{coarse_dim, coarse_dim}));
-    auto tmp = matrix_type::create(exec, gko::dim<2>{coarse_dim, fine_dim});
-    restrict_op->apply(amgxpgm_op, tmp.get());
-    tmp->apply(prolong_op.get(), coarse_matrix.get());
+    auto tmp = matrix_type::create(exec, gko::dim<2>{fine_dim, coarse_dim});
+    amgxpgm_op->apply(prolong_op.get(), tmp.get());
+    restrict_op->apply(tmp.get(), coarse_matrix.get());
```

Comment:
I'm curious about this change - it should compute the same result, does this have a performance impact?

Reply:
Yes, they give the same result in theory.
I think the parallel way usually depends on the #rows, so keeping more rows might be better.
the amgx_pgm paper claims it reduces the increasing hash table possibility if there are more rows.

Reply:
good point! This might improve parallelism for medium-sized problems. The rest of course depends on the SpGEMM implementation.

Reply:
I play around the current example I have, which is 6525 x 6525. It does not show the noticeable changes from this